base/compiler/ssair/inlining.jl

```diff
@@ -606,8 +606,12 @@ function rewrite_apply_exprargs!(ir::IRCode, idx::Int, argexprs::Vector{Any}, at
         # now push flattened types into new_atypes and getfield exprs into new_argexprs
         for j in 1:length(def_atypes)
             def_atype = def_atypes[j]
-            new_call = Expr(:call, Core.getfield, def, j)
-            new_argexpr = insert_node!(ir, idx, def_atype, new_call)
+            if isa(def_atype, Const) && is_inlineable_constant(def_atype.val)
+                new_argexpr = insert_node!(ir, idx, def_atype, quoted(def_atype.val))
```

Comment:
`Const` shouldn't need a node at all, right, just use the value as the argexpr?

Reply:
yes, I suspect that's true. I'll try it.